bcgov/HOUS-permit-portal
# Add support for building and location details page

## Description

- add seeds for jurisdiction HDD and use to provide default value for HDD and climate zone on part 3 checklist form
- add UI for building and location details page on part 3 checklist form along with necessary API updates

## What type of PR is this? (check all applicable)


## Related Tickets & Documents

[BPHH-2265](https://hous-bssb.atlassian.net/browse/BPHH-2265)
[BPHH-2442](https://hous-bssb.atlassian.net/browse/BPHH-2442)

## Steps to QA

1. Re-run migrations: 

- `rails db:migrate:down VERSION=20241030222657 && rails db:migrate:up VERSION=20241030222657`
- comment out change in `AddSectionCompletionStatusToPart3StepCodeChecklists`, then rerun: `rails db:migrate:down VERSION=20241127205040 && rails db:migrate:up VERSION=20241127205040`

2. run data migration: `rails data:migrate`
3. verify building and location details page on permit application page (/permit-applications/:id/edit/part-3-step-code/location-details). Details can be updated and are persisted and user is pushed to the next section (baseline occupancies) upon submission 
4. verify building and location details page on public page (/part-3-step-code/location-details). Details can be updated and user is pushed to the next section (baseline occupancies) upon submission . Note - currently a new step code is created on every page reload.

## UI Accessibility Checklist



_Note the following checklist is not an exhaustive list. Check
the [WebAIM checklist](https://webaim.org/standards/wcag/checklist) for
a user-friendly checklist with more details. Aim to meet at least *AA* conformance level where applicable. Check
the [WCAG guidelines](https://webaim.org/standards/wcag/checklist) for the full guidelines._

      with [Wave Browser Extension](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh)
      for accessibility errors?

## General Checklist


## [optional] Are there any post-deployment tasks we need to perform?

n/a


[BPHH-2265]: https://hous-bssb.atlassian.net/browse/BPHH-2265?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[BPHH-2442]: https://hous-bssb.atlassian.net/browse/BPHH-2442?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
